artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationPageWriteMessage.java

```diff
@@ -49,7 +49,7 @@ public int expectedEncodeSize() {
 
    @Override
    public void encodeRest(final ActiveMQBuffer buffer) {
-      buffer.writeInt(pageNumber);
+      buffer.writeInt((int)pageNumber);
```

Comment:
Ditto

Reply:
added versioning here, and it's using long.